tests/test_markdownviewer.md

```diff
@@ -0,0 +1,15 @@
+* [First](test_markdownviewer.md#first)
```

Comment:
Is the path required? Or anchors like `[First](#first)` will work too?

Reply:
Within the scope of this change, yes, the path is required as this is a change to the `load` method of `Markdown`, which doesn't really have the concept of "current document". Any such link would be coming in from application code that uses `Markdown`, so it should ensure that there is a patch.

But, you make a good point, it does suggest that `MarkdownViewer` should get a wee improvement to handle that case.

Reply:
Got it. Yeah I was reading the code, and currently this is what would happen with `#first`:

- clicking on `[First](#first)` will enter `MarkdownViewer.go("#first")`
- which will call `Navigator.go("#first")`
- which will return the current working directory suffixed with the anchor as a Path
- then `Markdown.load(Path.cwd() / "#first")` will split it again to CWD and `"first"`
- and will try to load CWD, which will fail

To fix this, `MarkdownViewer.go` should immediately sanitize the location, then either call `Markdown.load` if there's a path, or `Markdown._goto_anchor` if there's only an anchor.

BTW it would be nice if `_goto_anchor` was public :slightly_smiling_face: I'll probably have to use it myself, because the anchors generated by the Griffe-to-Markdown code contain dots, which will be removed by the new slugify functionality. So I'll have to catch the LinkClicked event to remove the dots as well before calling `_goto_anchor`

Reply:
Good call; I didn't really imagine anyone would want or need access to `_goto_anchor` but that makes a lot of sense. I'll make that change (and also tweak things in `MarkdownViewer` as we're discussing here too).